moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -323,6 +343,13 @@ bool PlanningPipeline::generatePlan(const planning_scene::PlanningSceneConstPtr&
         RCLCPP_ERROR(node_->get_logger(), "Planner '%s' failed", planner->getDescription().c_str());
         break;
       }
+
+      // TODO: should this be optional since the plugins already checked for timeout?
+      // check for timeout
+      if (std::chrono::duration<double>(clock::now() - planner_start_time).count() >= max_single_planner_time)
```

Comment:
Maybe I would even make sense to use the same check here as for the adapters
```cpp
if (std::chrono::duration<double>(clock::now() - plan_start_time).count() >= allowed_planning_time)
```
I think we might not care too much about the individual elements exceeding their allocated individual budget (for now) but for the overall planning process to not exceed the `allowed_planning_time` :thinking:

Reply:
Agreed. Especially because some planners will take longer, so splitting the timeout evenly seems risky.